### DISCUSSION TITLE
Dafny package as library

### DISCUSSION THREAD

**Discussion by hirataqdees** [dafny-lang/dafny#2247] [General]
Hi, 

I have two Dafny packages A and B. The package B heavily depends on the package A: types, functions, methods, verification contracts. Intuitively I would like to include A as a library to B. I find multiple prior and on-going discussions for adding Dafny packages as libraries to other packages, e.g: 
https://github.com/dafny-lang/dafny/issues/1236
https://github.com/dafny-lang/dafny/issues/2084

My question is: what is the state-of-the-art? are there any steps listed somewhere that will enable me to experiment with a basic example of including one package as a library to another? 

Thanks

**Comment by keyboardDrummer** [comment]
Currently there's no explicit support for libraries built into Dafny. One thing you can do is have your dependency, package A in your example, in a separate repository and then include that repository as a git submodule in the dependent package, package B in your example.

When invoking the Dafny CLI, make sure not to use `/verifyAllModules` if you don't want to verify the modules from your dependency, and don't pass the source files of your dependency on the CLI but instead use include directives in your dependent to use them.
